File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        cache = [0]*len(nums)
        cache[-1] = nums[-1]
        if len(nums) > 1:
            cache[-2] = max(nums[-1], nums[-2])
        else:
            return nums[0]
        for i in range(len(cache)-3, -1, -1):
            cache[i] = max(nums[i] + cache[i+2], cache[i+1])
        return cache[0]

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numset;
        vector<int> candidates;
        int res = 0;
        for (int num:nums){
            if (numset.count(num)){ continue; }
            else {
                numset.insert(num);
            }
        }
        for (int num:numset){
            if (numset.count(num-1)){ continue; }
            candidates.push_back(num);
        }
        int count=0;
        int val;
        for (int num:candidates){
            val = num;
            count = 1;
            while (numset.count(val+1)){
                val +=1;
                count +=1;
            }
            res = max(res, count);
        }
        return res;
    }
};
